mono/utils/mono-state.c

```diff
@@ -145,18 +145,44 @@ static const char *configured_timeline_dir;
 static MonoSummaryTimeline log;
 
 static void
-file_for_summary_stage (const char *directory, MonoSummaryStage stage, gchar *buff, size_t sizeof_buff)
+file_for_stage_breadcrumb (const char *directory, MonoSummaryStage stage, gchar *buff, size_t sizeof_buff)
 {
 	g_snprintf (buff, sizeof_buff, "%s%scrash_stage_%d", directory, G_DIR_SEPARATOR_S, stage);
 }
 
 static void
-create_stage_mark_file (void)
+file_for_signal_breadcrumb (const char *directory, const char *signal, gchar *buff, size_t sizeof_buff)
+{
+	g_snprintf (buff, sizeof_buff, "%s%scrash_signal_%s", directory, G_DIR_SEPARATOR_S, signal);
+}
+
+static void
+create_stage_breadcrumb (void)
 {
 	char out_file [200];
-	file_for_summary_stage (log.directory, log.level, out_file, sizeof(out_file));
+	file_for_stage_breadcrumb (log.directory, log.level, out_file, sizeof(out_file));
 	int handle = g_open (out_file, O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
-	close(handle);
+	if (handle < 0) {
+		g_async_safe_printf ("Failed to create breadcrumb file %s\n", out_file);
+		return;
+	}
+
+	if (close(handle) < 0)
+		g_async_safe_printf ("Failed to close breadcrumb file %s\n", out_file);
+}
+
+static void
+create_signal_breadcrumb (const char *signal)
+{
+	char out_file [200];
+	file_for_signal_breadcrumb (log.directory, signal, out_file, sizeof(out_file));
+	int handle = g_open (out_file, O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
+	if (handle < 0) {
+		g_async_safe_printf ("Failed to create breadcrumb file %s\n", out_file);
+		return;
+	}
+	if (close(handle) < 0)
+		g_async_safe_printf ("Failed to close breadcrumb file %s\n", out_file);
```

Comment:
I'd like to see this logic shared in a helper function between `create_stage_breadcrumb` and `create_signal_breadcrumb`.